File: examples/csharp/printsupported.cs
/**
 * @file    printsupported.cs
 * @brief   Prints all SBML Levels and Versions supported by this version 
 *          of libsbml.
 * @author  Frank Bergmann
 *
 *
 * This file is part of libSBML.  Please visit http://sbml.org for more
 * information about SBML, and the latest version of libSBML.
 */

using System;
using libsbmlcs;

public static class PrintSupported
{
    /// <summary>
    /// The main entry point for the application.
    /// </summary>
    [STAThread]
    private static void Main(string[] args)
    {	
        Console.WriteLine("Supported by LibSBML " + 
					libsbml.getLibSBMLDottedVersion());
		Console.WriteLine();
		
		SBMLNamespacesList supported = SBMLNamespaces.getSupportedNamespaces();
		for (uint i = 0; i < supported.getSize(); i++)
		{
			SBMLNamespaces current = supported.get(i);
			Console.WriteLine("\tSBML Level: " + current.getLevel() + 
							" Version: " + current.getVersion());				
		}
		
		Console.WriteLine();
    }
}

File: examples/python/printsupported.py
#!/usr/bin/env python
## 
## @file    printsupported.cs
## @brief   Prints all SBML Levels and Versions supported by this version 
##          of libsbml.
## @author  Frank Bergmann
## 
## 
## This file is part of libSBML.  Please visit http://sbml.org for more
## information about SBML, and the latest version of libSBML.
## 

import sys
import os.path
from libsbml import *

print("Supported by LibSBML " + getLibSBMLDottedVersion());
print;

supported = SBMLNamespaces.getSupportedNamespaces();
for i in range(0,supported.getSize()):
	current = supported.get(i);
	print("\tSBML Level: " + str(current.getLevel()) + 
					" Version: " + str(current.getVersion()));
print;



